doc/developer/design/20220214_create_cluster.md

```diff
@@ -0,0 +1,191 @@
+# Initial cluster SQL API
+
+## Summary
+
+This document proposes the initial SQL API changes to introduce
+[clusters](../platform/ux.md#cluster) to Materialize.
+
+## Design
+
+### Cluster SQL syntax and semantics
+
+Four new SQL statements will be introduced with the following grammar to
+create, alter, drop, and list clusters, respectively:
+
+```
+create_cluster_stmt ::=
+  CREATE CLUSTER [IF NOT EXISTS] <name> [[WITH] <option> [, <option>...]]
+
+alter_cluster_stmt ::=
+  ALTER CLUSTER <name> [[SET] <option> [, <option>...]]
+
+drop_cluster_stmt ::=
+  DROP CLUSTER [IF EXISTS] <name> [CASCADE]
+
+show_cluster_stmt: SHOW CLUSTERS [{ LIKE 'pattern' | WHERE <expr> }]
+
+<option> ::=
+      VIRTUAL
+    | SIZE '<size>'
+
+<name> ::= <unqualified-identifier>
+```
+
+**All of these statements will be available in experimental mode only.**
+
+The name of a cluster is an unqualified identifier. Clusters are scoped to
+a catalog, and so do not belong in the usual namespace of databases and schemas.
+For a similarly scoped object in PostgreSQL, see [publications].
+
+The `CREATE CLUSTER` statement creates a new cluster. It returns an error if
+a cluster with the specified name already exists unless the `IF NOT EXISTS`
+clause is present.
+
+The `ALTER CLUSTER` statement changes properties of an existing cluster.
```

Comment:
Is the intention to allow users to change the `SIZE` of a cluster dynamically? In practice is the entire `ALTER` feature going to return an error for anything you try to change for a long time? Seems also weird to change between `VIRTUAL` and `SIZE`.

Reply:
> Is the intention to allow users to change the SIZE of a cluster dynamically? 

Yes, exactly.

> In practice is the entire ALTER feature going to return an error for anything you try to change for a long time? 

Nope!

> Seems also weird to change between VIRTUAL and SIZE.

Agreed. If these virtual clusters end up sticking around though, could make sense? @frankmcsherry thought they might be useful as a way of partitioning up indexes across some theoretical boundaries, and then only introducing the physical resources via `ALTER` when you're ready to spend the $$.